### PR TITLE
Simplify CodegenCppVisitor

### PR DESCRIPTION
**Pull request by iomaganaris**
- Rename C to C++ in multiple places
- Remove wrapper_printer and target_printer since we only print a single C++ file now for all the backends (C++, OpenACC/OpenMP). There is no more ISPC which needed the wrapper.

### CODE REVIEW

File: src/codegen/codegen_cpp_visitor.hpp [hunk 1]
```diff
@@ -194,17 +194,7 @@ class CodegenCppVisitor: public visitor::ConstAstVisitor {
     /**
      * Code printer object for target (C++)
      */
-    std::shared_ptr<CodePrinter> target_printer;
-
-    /**
-     * Code printer object for wrappers
-     */
-    std::shared_ptr<CodePrinter> wrapper_printer;
```
Comment (pramodk) [inline src/codegen/codegen_cpp_visitor.hpp:202]: Oh yes! Have very "good" memories about this! 🙃

**Review by pramodk — APPROVED** [review]
LGTM

**Review by tristan0x — APPROVED** [review]
LGTM!
Sorry for the conflict @iomaganaris ¯\_(ツ)_/¯

**Comment by iomaganaris** [comment]
> LGTM! Sorry for the conflict @iomaganaris ¯_(ツ)_/¯

Haha no worries, it was nothing serious :innocent:

**Review by alkino — APPROVED** [review]

**Pull request merged**